source/templates/config.js

```diff
@@ -30,6 +30,11 @@ function updateProviderFields(isChangeEvent = false) {
     } else {
         setElementDisplay('tmdb-fields', 'none');
     }
+    if (!document.getElementById('get-all-languages')?.checked) {
+        setElementDisplay('languages-fields', 'block');
+    } else {
+        setElementDisplay('languages-fields', 'none');
+    }
```

Comment:
Hi @Ariurn, the problem you solved is related to my issue #176. 

However, in your solution you go the extra mile and hide the language selection field (see pinned line of code).

While it might make sense in your situation, that prevents users from leveraging the translated title search with TMDB metadata as self.config['languages'] will be empty.

Ideally, we should detect indexers' languages before collecting TMDB meta (see [jackett_service.py#L192-L201](https://github.com/aymene69/stremio-jackett/blob/main/source/jackett/jackett_service.py#L192-L201)).

Then, we could create a branch in [tmdb.py#L16](https://github.com/aymene69/stremio-jackett/blob/main/source/metdata/tmdb.py#L17) like

```py
        if self.config['getAllLanguages']:
          languages = [indexer.language for indexer in self._indexers]
        else:
          languages = self.config['languages']
        for lang in languages:
```

This way, when we check for for languages in [jackett_service.py#L133-L137](https://github.com/aymene69/stremio-jackett/blob/main/source/jackett/jackett_service.py#L133-L137), series.languages will be the same as the languages of our Jackett's configured indexers.

```py
        else:
            index_of_language = [index for index, lang in enumerate(series.languages) if
                                 lang == indexer.language or lang == 'en']
            languages = [series.languages[index] for index in index_of_language]
            titles = [series.titles[index] for index in index_of_language]
```

In fact, that would drive the language selection useless as it would be much more powerful to use the getAllLanguages checkbox.

An easier (and hopefully temporary) solution would be to just let the languages-fields remain visible when TMDB metadata is selected. Instead of disabling it, you could just add a disclaimer (maybe in a [bootstrap danger alert](https://getbootstrap.com/docs/4.0/components/alerts/#examples)) that selected languages will be only used for translated title search.

Reply:
@farebyting @aymene69 What do you think of my suggestions?